accelerator/config.c

```diff
@@ -122,6 +147,100 @@ status_t ta_config_default_init(ta_config_t* const info, iota_config_t* const ic
   return ret;
 }
 
+status_t ta_config_file_init(ta_core_t* const conf, int argc, char** argv) {
+  int key = 0;
+  status_t ret = SC_OK;
+  struct option* long_options = cli_build_options();
+  yaml_parser_t parser;
+  yaml_token_t token;
+  FILE* file = NULL;
+  char* arg = NULL;
+  int state = 0;
+
+  // Initialize default configuration file path with '\0'
+  conf->conf_file[0] = '\0';
+
+  if (!yaml_parser_initialize(&parser)) {
+    ret = SC_CONF_PARSER_ERROR;
+    log_error(config_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_CONF_PARSER_ERROR");
+    goto done;
+  }
+
+  // Looping through the CLI arguments a first time to find the configuration file path
+  while ((key = getopt_long(argc, argv, "hv", long_options, NULL)) != -1) {
+    switch (key) {
+      case ':':
+        ret = SC_CONF_MISSING_ARGUMENT;
+        log_error(config_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_CONF_MISSING_ARGUMENT");
+        break;
+      case '?':
+        ret = SC_CONF_UNKNOWN_OPTION;
+        log_error(config_logger_id, "[%s:%d:%s]\n", __func__, __LINE__, "SC_CONF_UNKNOWN_OPTION");
+        break;
+      case CONF_CLI:
+        ret = cli_config_set(conf->conf_file, &conf->info, &conf->iconf, &conf->cache, &conf->service, key, optarg);
+        break;
+      default:
+        break;
+    }
+    if (ret != SC_OK) {
+      break;
+    }
+  }
+
+  // Resetting the CLI option index for the second loop where they are actually analyzed
+  optind = 1;
+
+  if ((file = fopen(conf->conf_file, "r")) == NULL) {
+    ret = SC_OK;
```

Comment:
Why does here return `SC_OK` if fopen fails?
Maybe you could first check on the length of `conf_file`, then decide to fopen or not.